Sources/AppcuesKit/Presentation/ExperienceRendering/ExperienceStateMachine+State.swift

```diff
@@ -152,9 +152,22 @@ extension ExperienceStateMachine.Transition {
         let stepIndex = Experience.StepIndex.initial
         do {
             let package = try traitComposer.package(experience: experience, stepIndex: stepIndex)
+
+            // for pre-step navigation actions - only allow these to execute if this experience is being launched for some
+            // other reason than qualification (i.e. deep links, preview, manual show). For any qualified experience, the initial
+            // starting state of the experience is determined solely by flow settings determining the trigger
+            // (i.e. trigger on certain screen).
+            let navigationActions: [Experience.Action]
+            if case .qualification = experience.trigger {
```

Comment:
I don't usually like using non-exhaustive conditionals for enums since it can be  hard to know what needs to be updated if we added another `ExperienceTrigger` case, especially when it's not a property of the enum itself. So for this, I think a computed property on `ExperienceTrigger` like `shouldProcessNavigation` would work nicely.

Reply:
Good call - agree that is a better approach, updated.